src/utils/get-server-status.ts

```diff
@@ -0,0 +1,180 @@
+import { Config } from './../interfaces/config'
+import { sendAlerts } from './alert'
+import { ServerStatus, StatusCount } from './../interfaces/server-status'
+import { Probe } from '../interfaces/probe'
+import { ValidateResponseStatus } from './alert'
+import console, { log } from 'console'
+
+const SERVER_STATUSES: ServerStatus[] = []
+
+// Function to calculate successes and failures, either total or consecutives.
+// If consecutive success/failure equals to threshold, it will send notifications
+const calculateSuccessFailure = async ({
+  errorName,
+  probeStatus,
+  validation,
+  config,
+  request,
+  threshold,
+}: {
+  errorName: string
+  probeStatus: StatusCount
+  validation: ValidateResponseStatus
+  config: Config
+  request: Probe['request']
+  threshold: number
+}) => {
+  log(`Checking for ${errorName} in probe...`)
+  if (probeStatus.isDown === false) {
```

Comment:
Kindly reduce nested ifs in here.

Reply:
It's kinda hard for me to reduce nested ifs. I use switch case. Is it okay?